### DISCUSSION TITLE
Leveraging Directus Vue Components in a Custom Display Extension

### DISCUSSION THREAD

**Discussion by mark-james** [directus/directus#6875] [Archive]
Hi,
I've never used vue beyond a few basic hello world examples, so this may have a simple answer. How do I use the built-in Directus vue components within a custom display component?

I've successfully built the example in the docs. 

```
<template>
	<div>Value: {{ value }}</div>
</template>

<script>
export default {
	props: {
		value: String,
	},
};
</script>
```

But if I change it to use a button like this -

```
<template>
	<v-button>Click me!</v-button>
</template>

<script>
export default {
	props: {
		value: String,
	},
};
</script>
```

It doesn't show up in the table. What am I doing wrong?

Thanks
Mark

**Comment by rijkvanzanten** [comment]
@nickrum ?

**Comment by nickrum** [comment]
Hi @mark-james, that is exactly how you use Directus' built-in components. They are globally registered and vue will resolve them automatically.

The snippet you provided works perfectly fine over here:

![click](https://user-images.githubusercontent.com/12128505/126187506-c92a5bd6-c03c-443e-bf09-a40f8bd3202b.png)

Does the extension show up in the displays list when configuring a field?

**Reply by mark-james** [reply]
Hi Nick,
I am able to configure it. But I'm not seeing the button render. It's blank. 

![Screen Shot 2021-07-20 at 8 58 21 am](https://user-images.githubusercontent.com/26987752/126238221-25ff534b-fe01-49e2-a4ad-50268cf8de38.png)

Inspecting the cell, I can see a v-button tag in there. 
![Screen Shot 2021-07-20 at 9 04 02 am](https://user-images.githubusercontent.com/26987752/126238401-81f32449-3c83-4e69-aa06-b0f882f8259d.png)

**Reply by nickrum** [reply]
Looks like vue could indeed not find the `v-button` component. Could you share the extension source that is imported by the Directus app? It can be found under `<PUBLIC_URL>/extensions/displays/index.js`.

**Reply by mark-james** [reply]
Index.js attached.
[index.js.zip](https://github.com/directus/directus/files/6853233/index.js.zip)

**Reply by nickrum** [reply]
Looks like vue has been bundled into your extension. Are you using the extension-sdk to build your extension?

**Reply by mark-james** [reply]
Hi Nic,
I was using rollup directly as I was having trouble with the extension. I've just gone back to a fresh project extension had another shot (copying the doc instructions verbatim). However, I get this error when I trying to build -

[Error] Current directory is not a Directus extension.

Taking a quick peek at the code, it looks like an issue with my package.json? But I'm not sure what.  Any suggestions?

cheers
Mark

**Reply by mark-james** [reply]
Hi @nickrum,
Looking further into the extension, it looks like you're adding functionality right now. Is that correct?

The docs don't seem complete to get a minimum extension project up and running.

So I can keep moving while you lock things down, is it possible to get an example boilerplate display extension that works? So I can just add to it?

Thanks
Mark

**Reply by nickrum** [reply]
Hey @mark-james, sorry for the delay. I just added a note to the [docs](https://docs.directus.io/guides/displays/#_2-install-dependencies) on how to get around the `[Error] Current directory is not a Directus extension.` issue. You have to add a specific field to the `package.json` file for the extension CLI to pick up the extension type and input/output paths.

Yes you are right, extensions are heavily being worked on and there is more to come within the next weeks/months.
